### ISSUE TITLE
Infinite drops?

### ISSUE THREAD

**Issue by PaulRosenfield** [HabitRPG/habitica#2049]
I see from the other bug reports that you are having problems with checking and unchecking dailies, but I don't think anyone has mentioned this one specifically.

It seems like each time I check a daily, I have a chance at getting a new drop, but I don't lose the old drop when I uncheck a daily. Thus, by checking and unchecking, I end up with many drops for the same daily. Kind of a minor issue, since there is no real reason why anyone would regularly do this except to abuse the system, but I thought I'd let you know about it.

I'm using Chrome.

**Comment by wc8** [comment]
My impression is the fix will be harder than it’s worth, as the system would have to remember what item dropped from what task. I suspect it doesn't currently. Perhaps if the drop algorithm can be enhanced, the chance of a drop might be reduced making it less likely that clicking the same task again would give you a new drop.

**Comment by PaulRosenfield** [comment]
I'm inclined to agree: probably not worth fixing for its own sake. Just thought I'd share in case it helped you to understand your program's checking/unchecking behavior in general.

**Comment by Evxie** [comment]
When checking and unchecking you lose more exp/gold when unchecking than you gained when checking it. Atleast that's the way it's been for me. So by checking and unchecking I end up with 1exp less and 2silver less than I had before. I think that would be pretty demotivational for people trying to abuse it for drops.

**Comment by BlackCompany** [comment]
I noticed this bug as well except the xp adjustment is usually correct and the gold I loose less than I gained so if repeated gold can be increased albeit at a very slow pace and occasionally takes a big negative and gets closer to the original starting point. The xp can increase slightly when for some reason the task advances in color. I cant remember the term used for it. The task if checked and unchecked although the streak count stays correct color will advance and adjust the xp amount. For some reason this leads to occasionally giving more xp than it takes away but again a very small amount. I also experienced the item drops. Of course as mentioned this only matters if someone wants to abuse the system. I just have a bad habit of wanting to know what causes something and experimented with it to see what values changed etc.
